components/ecodan/response.cpp

```diff
@@ -175,18 +175,24 @@ namespace ecodan
             case GetType::PUMP_STATUS:
                 // byte 1 = pump running on/off
                 // byte 4 = pump 2
+                // byte 5 = pump 3
                 // byte 6 = 3 way valve on/off
                 // byte 7 - 3 way valve 2            
+                // byte 11 - Mixing valve status
                 status.WaterPumpActive = res[1] != 0;
                 status.ThreeWayValveActive = res[6] != 0;
                 status.WaterPump2Active = res[4] != 0;
-                status.ThreeWayValve2Active = res[7] != 0;                
+                status.ThreeWayValve2Active = res[7] != 0;         
+                status.WaterPump3Active = res[5] != 0;       
+                status.MixingValveStatus = res[11];   
                 publish_state("status_water_pump", status.WaterPumpActive);
                 publish_state("status_three_way_valve", status.ThreeWayValveActive);
                 publish_state("status_water_pump_2", status.WaterPump2Active);
-                publish_state("status_three_way_valve_2", status.ThreeWayValve2Active);                
+                publish_state("status_three_way_valve_2", status.ThreeWayValve2Active);
+                publish_state("status_water_pump_3", status.WaterPump3Active);
+                publish_state("status_mixing_valve", status.MixingValveStatus);

```

Comment:
If we leave this as an internal variable, you can use something like https://github.com/gekkekoe/esphome-ecodan-hp/blob/5d0efaea72b55d9e758db12db55610d70f8ea90f/confs/base.yaml#L129 to map it internal value to a readable string

Reply:
for 0x15, byte 11, other system are reporting 04 there.

Byte 11 - Mixing valve (out5) : 0:idle 1:closing 2:opening 3:close/stop
You did not see this value ?

Reply:
I've created text sensor in confs/zone2.yaml (line 85) but for some reason it always returns "Idle" - currently I have both sensors and I see that value vas changed, but this text sensor was not updated...

Reply:
Hmm is it possible that "refresh rate" for this text sensor is too low:
![image](https://github.com/user-attachments/assets/1cf313a2-ff14-4fec-bad2-fffcfeb006ad)
The 3rd change was detected - but duration is for whole minute, and these corrections usually last for only about 10-15s. If this is the case then it would be best if we would send the text code directly from c++ part right?

Reply:
Ah, so we only query 0x15 once every 21 seconds. So it would not matter if done via c++ directly. Since the 0x15 response is only retrieve once every 21 seconds. you can try to reduce the polling to 500ms (https://github.com/gekkekoe/esphome-ecodan-hp/blob/5d0efaea72b55d9e758db12db55610d70f8ea90f/components/ecodan/__init__.py#L27) but that would give you once every 10.5 seconds.

I do generally emulate what the procon and melcloud are doing, they also seem to query 1 topic per second, but we might need to tweak it

Reply:
> I've created text sensor in confs/zone2.yaml (line 85) but for some reason it always returns "Idle" - currently I have both sensors and I see that value vas changed, but this text sensor was not updated...

you can try to publish the template sensor, it will trigger the rendering. you can do that when the value was changed

Reply:
That is not so important/critical property so I don't think that this justifies increased polling. Another example:
![image](https://github.com/user-attachments/assets/8e0ece93-180f-42e2-a068-64dda2ac81b5)

Reply:
agreed, would like to keep it as close as the official adapters as possible

Reply:
> Hmm is it possible that "refresh rate" for this text sensor is too low: ![image](https://private-user-images.githubusercontent.com/44526436/389257722-1cf313a2-ff14-4fec-bad2-fffcfeb006ad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI0NDA2NzAsIm5iZiI6MTczMjQ0MDM3MCwicGF0aCI6Ii80NDUyNjQzNi8zODkyNTc3MjItMWNmMzEzYTItZmYxNC00ZmVjLWJhZDItZmZmY2ZlYjAwNmFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI0VDA5MjYxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwMzM5ZTI5Zjc5M2Q2NGE5OTY5YTIzMzU4Yjc1ZjUyOTQ3ODg5Yzc0NmM0MjU3MDMzNjZjYjMyNGQ0ZTJiODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Y7_jRD-ja4vLC0Cth6SSlxN539Pmj02tQ2Rrqiq8AtY) The 3rd change was detected - but duration is for whole minute, and these corrections usually last for only about 10-15s. If this is the case then it would be best if we would send the text code directly from c++ part right?

I think this is similar to the defrost stages - i have run polling down to ~350ms stable per msg, but issues were reported before for some bytes slow

@DamijanD - Can you verify multi-zone working from 0x03 please:
![image](https://github.com/user-attachments/assets/98dcbe4c-71ca-4f21-8cc1-e245e0d0cc51)

Reply:
@F1p currently I get 1 which corresponds with real situation. I will keep an eye on it :-)

Reply:
nice, but lets try not to add more scope to this PR :) Let's have this tested and merged.

Reply:
> Mixing valve

Byte 10 = Mixing Step in procon
![image](https://github.com/user-attachments/assets/f8fcab61-47a1-470f-9ce0-2c2283df250b)

Unable to locate Byte 11 currently

Reply:
> nice, but lets try not to add more scope to this PR :) Let's have this tested and merged.
Agree I cleaned up the code. (but I will make new PR with this change after this one is merged because it is something that is nice to have in multi zone setup).
I also removed the text part for mixing valve status as nothing I did was responsive enough (and we don't want to mapping to text status in response.cpp because of multi language).

Reply:
new PR are always appreciated. And napping in c++ code would not make any difference. It's only refreshed from the heatpump once every 21s